### DISCUSSION TITLE
Is there a new process hierarchy change after WebView2 runtime version 117.0*?

### DISCUSSION THREAD

**Discussion by tamuseanmiller** [MicrosoftEdge/WebView2Feedback#3848] [Q&A] [answered]
Hello-

We've been using WebView2 for a while now and after this most recent update, we've noticed that some of our functionality doesn't work. Previously the process hierarchy was rooted in the process that owned the WebView2 control which was very handy, here's a screenshot of what Outlook currently looks like:
![image](https://github.com/MicrosoftEdge/WebView2Feedback/assets/42820988/eef45cb3-f7d3-45ca-9497-699ecbb7a08c)

This is what we want to happen but instead, we're seeing WebView2 rooted in Explorer.exe. Is there any way to use the legacy process hierarchy?

**Comment by victorhuangwq** [comment]
@lflores-ms could you take a look at this?

**Comment by champnic** [comment, marked as answer]
Hey @tamuseanmiller - It's possible this was due to a change to avoid a bug when DPI compatibility settings were causing launch failures when trying to parent the processes/HWNDs. Instead, the child processes get launched from explorer.exe. You could check if your parent process has a "__COMPAT_LAYER" environment variable set, which could indicate this is happening. I also believe we use explorer.exe as the parent process if the host app is running as admin, as we try to avoid running the WebView2 processes as admin for security.

**Reply by tamuseanmiller** [reply]
Hey @champnic - thanks for taking a look. I don't believe we have the **parent** of the WebView2 process launched as admin in this state but our process that launches the WebView2 control does have the `HIGHDPIAWARE` registry key set. Do you have any further idea as to how this can be avoided?

**Reply by tamuseanmiller** [reply]
As for why we use `HIGHDPIAWARE` in that process, we see weird behaviors with WebView2 when using low scaling. Basically, the WebView2 browser would overlap onto the dialog we have it contained in. I can't remember if we reproduced this with the sample code or not.

**Reply by tamuseanmiller** [reply]
Just want to add to this after further testing. The above comments don't seem to have an effect. I found that we're adding settings to our manifest files like this:
```xml
<application xmlns="urn:schemas-microsoft-com:asm.v3">
    <windowsSettings>
	<dpiAwareness xmlns="http://schemas.microsoft.com/SMI/2016/WindowsSettings">PerMonitor</dpiAwareness>
        <dpiAware  xmlns="http://schemas.microsoft.com/SMI/2005/WindowsSettings">true</dpiAware>
    </windowsSettings>
</application>
```

Ideally, we would want to keep these as is but still have the same legacy functionality from before this bug fix. The changes that will need to be done to fix this on our end would be sizeable and this is already affecting many of our customers. Is there anything further we can do without reverting these settings @champnic ?

**Reply by champnic** [reply]
We generally recommend using PerMonitorV2 DPI awareness. Is that something you can try for your app, if you are already PerMonitor? One of the listed changes between the 2 versions is automatic support for dialogs, which may be relevant to the issue you mentioned seeing: https://learn.microsoft.com/en-us/windows/win32/hidpi/high-dpi-desktop-application-development-on-windows
<img width="297" alt="image" src="https://github.com/MicrosoftEdge/WebView2Feedback/assets/45700886/509efc66-a334-4610-b51f-175e17cebff5">

**Reply by tamuseanmiller** [reply]
I assume that this means if we switch to using PerMonitor v2 then this won't interact with your DPI bug fix and we may see the process hierarchy we're looking for?

**Reply by champnic** [reply]
I believe you would also need to remove the HIGHDPIAWARE registry key value, though with PerMonitorV2 that shouldn't be necessary.

**Reply by tamuseanmiller** [reply]
Thanks for all your help here, I think I have a possible fix now. We wouldn't have been able to figure this out without your help. Really appreciate it!

**Reply by champnic** [reply]
Great, glad to hear you have a path forwards!
